pkg/gardenlet/controller/shoot/shoot_control.go

```diff
@@ -546,10 +547,12 @@ func (c *Controller) updateShootStatusOperationStart(ctx context.Context, g gard
 		})
 }
 
-func (c *Controller) updateShootStatusOperationSuccess(ctx context.Context, g gardencore.Interface, shoot *gardencorev1beta1.Shoot, seedName string, operationType gardencorev1beta1.LastOperationType) (*gardencorev1beta1.Shoot, error) {
+func (c *Controller) updateShootStatusOperationSuccess(ctx context.Context, g gardencore.Interface, shoot *gardencorev1beta1.Shoot, shootSeedNamespace string, seedName string, operationType gardencorev1beta1.LastOperationType) (*gardencorev1beta1.Shoot, error) {
 	var (
-		now         = metav1.NewTime(time.Now().UTC())
-		description string
+		now          = metav1.NewTime(time.Now().UTC())
+		description  string
+		err          error
+		isHibernated bool
```

Comment:
```suggestion
		isHibernated = gardencorev1beta1helper.HibernationIsEnabled(shoot)
```

Reply:
This unfortunately causes `gardencorev1beta1helper.HibernationIsEnabled(shoot)` to always be executed even though it is not required if `len(shootSeedNamespace) > 0`. Thats why it is in an if-else block.

Reply:
Yes, but it's not a costly operation at all, so doing it here will save some nesting below (improving readability slightly). I don't insist on this, but I generally think it's a better readable style to do it like this in case the operation is cheap.

Reply:
ok sure not problem - its indeed only a minimal gain. I though about it as well - but I see it is an issue for you so I ll change it.